integration-test/src/test/java/com/datastrato/gravitino/integration/test/container/BaseContainer.java

```diff
@@ -124,7 +125,33 @@ public void start() {
     container.start();
   }
 
-  protected abstract boolean checkContainerStatus(int retryLimit);
+  protected abstract boolean checkContainerStatus(int timeoutMillis);
+
+  protected boolean checkContainerStatusWithRetry(int timeoutMillis, BooleanSupplier... checker) {
```

Comment:
Do we support multiple checks? If so, should we only perform them in serialization?

Reply:
Yes, such as the Hive container, we need to check the HDFS data node, HMS client connection and HDFS client connection